airflow/jobs.py

```diff
@@ -2023,8 +2017,10 @@ def _execute(self):
                                                'scheduler_zombie_task_threshold')
             while True:
                 # Monitor the task to see if it's done
-                return_code = self.process.poll()
+                return_code = self.task_runner.return_code()
                 if return_code is not None:
+                    self.logger.info("Task exited with return code {}"
```

Comment:
logger.debug (assuming 0 is not None). Maybe better to log > 0 to info (or even just keep that at debug as airflow should handle that)?